### PR TITLE
Convert publish.js to Ramda functions

### PR DESCRIPTION
**Pull request by MattMS**

### CODE REVIEW

File: publish.js [hunk 1]
```diff
@@ -1,93 +1,109 @@
...
-    }),
+var simplifySee = R.pipe(
+  R.map(
+    R.pipe(
+      R.split(/\s*,\s*/),
+      R.map(R.replace(/^R[.]/, ''))
+    )
+  ),
+  R.flatten
+)
```
Comment (davidchambers) [inline publish.js]: Here's a simpler version:

```javascript
//  simplifySee :: Array String -> Array String
var simplifySee =
R.pipe(R.chain(R.split(/\s*,\s*/)),
       R.map(R.replace(/^R[.]/, '')));
```

Comment (MattMS) [inline publish.js]: Nice, thanks!
Updated PR.

**Comment by davidchambers** [comment]
I left one more comment, but this looks great!

File: publish.js [hunk 2]
```diff
@@ -1,93 +1,101 @@
...
-  return result;
-};
 
-var prettifySig = function(s) {
-  return s.replace(/[.][.][.]/g, '\u2026').replace(/->/g, '\u2192');
-};
+var prettifyCode = R.pipe(
+  R.join('\n'),
+  R.replace(/^[ ]{5}/gm, ''),
+  R.flip(R.invoker(2, 'highlight')('javascript'))(hljs),
```
Comment (MattMS) [inline publish.js]: Yeah, I did have mixed feelings about that code.
I've used a fat arrow instead, hope that's still fine.

Comment (davidchambers) [inline publish.js]: If we're to use a fat arrow here, why not throughout the file? While we're at it we could replace occurrences of `var` with `const`. :)

Comment (MattMS) [inline publish.js]: Quick find-replace, no problems!

I had just wanted to make sure ES6 was fine first.

**Comment by MattMS** [comment]
Is this all good now?
I think I fixed that extra comment you made.

**Comment by buzzdecafe** [comment]
i haven't found time to run it locally yet, but if it passed the gauntlet of @davidchambers review, i would expect it's good

**Comment by davidchambers** [comment]
LGTM

**Comment by buzzdecafe** [comment]
:cow2:

**Pull request merged**